src/functions.R

```diff
@@ -227,3 +267,107 @@ fetch_wqp_data <- function(site_counts_grouped, char_names, wqp_args = NULL,
   return(wqp_data_out)
 }
 
+
+#' @title Export a single target to Google Drive
+#' 
+#' @description
+#' A function to export a single target (as a file) to Google Drive and return
+#' the shareable Drive link as a file path.
+#' 
+#' @param target The name of the target to be exported (as an object not a string).
+#' 
+#' @param folder_pattern A local path to a location where the target should be
+#' exported before uploading.
+#' 
+#' @param drive_path A path to the folder on Google Drive where the file
+#' should be saved.
+#' 
+#' @returns 
+#' Returns a local path to a csv file containing a text link to the uploaded
+#' file in Google Drive.
+#' 
+export_single_file <- function(target, drive_path){
+  
+  require(googledrive)
+  
+  # Get target name as a string
+  target_string <- deparse(substitute(target))
+  
+  # Create a temporary file exported locally, which can then be used to upload
+  # to Google Drive
+  file_local_path <- tempfile(fileext = ".rds")
+  
+  write_rds(x = target,
+            file = file_local_path)
+  
+  # Once locally exported, send to Google Drive
+  out_file <- drive_put(media = file_local_path,
+                        # The folder on Google Drive
+                        path = drive_path,
+                        # The filename on Google Drive
+                        name = paste0(target_string, ".rds"))
+  
+  # Make the Google Drive link shareable: anyone can view
+  out_file_share <- out_file %>%
```

Comment:
suggestion: these two lines could be condensed as the following:

```suggestion
drive_share(out_file, role = "reader", type = "anyone")
```